cpp/experiments/employee_factory/source/tester.h

```diff
@@ -0,0 +1,15 @@
+#ifndef EMPLOYEE_FACTORY_TESTER_H
```

Comment:
You can also use `#pragma once` instead of include guards. It is not part of the cpp standard but it is supported by vast majority of modern compiler. It's fairly used in practice but it's up to you.

Reply:
Done

Reply:
I cant give you a concrete example. Just imagine you are using some outdated or obscure compiler.

And for this protection against duplicate files: 
if you have:
```
/include/lib1/file.hpp
/include/lib2/file.hpp
```
where both files are *identical*,
in the case of pragma once - they are considered as two separate/different files and including them both might cause an error.
but in the case of include guards, since both files are identical and include guards will clash and prevent double definitions

so in `/src/main.cpp`
```c++
#include <lib1/file.hpp>
#include <lib2/file.hpp>
```
pragma once = compile time error due to duplicate definitions
include guards = all ok

This is something i never considered, thnx for mentioning it, learned something new. 

So if you want to protect against that case you might want include guard. But if you have duplicate definition then you should probably refactor your code to avoid code duplication.

Reply:
If they have the same content, why would it work with `header guards` but not with `#pragma once`?

Preprocessor replaces header anyway. I would expect it to clash in both cases.

Does the header guard add some wrapping around the header definition so it is fine?